src/dtype.rs

```diff
@@ -503,16 +490,14 @@ pub trait PyArrayDescrMethods<'py>: Sealed {
     }
 
     /// Returns true if the type descriptor is a sub-array.
-    fn has_subarray(&self) -> bool {
-        // equivalent to PyDataType_HASSUBARRAY(self)
-        unsafe { !(*self.as_dtype_ptr()).subarray.is_null() }
-    }
+    ///
+    /// Equivalent to PyDataType_HASSUBARRAY(self)
```

Comment:
Including this in the documentation is nice, but please turn it into a link in the NumPy docs. (Same for `PyData_HASFIELDS`.)

Reply:
These don't appear to be documented anywhere. We could link to the macro definitions in the source if desired:https://github.com/numpy/numpy/blob/b3ddf2fd33232b8939f48c7c68a61c10257cd0c5/numpy/_core/include/numpy/ndarrayobject.h#L254-L255